source/Content.ts

```diff
@@ -0,0 +1,200 @@
+import { components } from '@octokit/openapi-types';
+import { Filter, ListModel, Stream, toggle } from 'mobx-restful';
+import { makeArray } from 'web-utility';
+
+import { githubClient } from './client';
+
+export type ContentItem = components['schemas']['content-directory'][number];
+export type ContentFile = components['schemas']['content-file'];
+
+export interface ContentFilter extends Filter<ContentItem> {
+    parent_path?: string;
+}
+
+export interface ContentItemWithPath extends ContentItem {
+    full_path?: string;
+}
+
+export class ContentModel extends Stream<ContentItemWithPath>(ListModel) {
+    client = githubClient;
+
+    constructor(
+        public owner: string,
+        public repository: string
+    ) {
+        super();
+        this.baseURI = `repos/${owner}/${repository}/contents`;
+    }
+
+    /**
+     * Get repository content at a specific path
+     * @see {@link https://docs.github.com/en/rest/repos/contents#get-repository-content}
+     */
+    @toggle('downloading')
+    async getOne(path: string) {
+        const { body } = await this.client.get<ContentItem | ContentItem[]>(
+            `${this.baseURI}/${path}`
+        );
+        return Array.isArray(body) ? body[0] : body!;
+    }
+
+    async *openStream({ parent_path }: ContentFilter) {
+        const setCount = (total: number) => (this.totalCount = total);
+
+        yield* parent_path
+            ? this.traverseChildren(parent_path, setCount)
+            : this.traverseTree(undefined, setCount);
+    }
+
+    /**
+     * Get direct children of a directory
+     */
+    async *traverseChildren(
+        parent_path = '',
+        onCount: (total: number) => any = () => {}
+    ): AsyncGenerator<ContentItemWithPath> {
+        try {
+            const { body } = await this.client.get<ContentItem[]>(
+                `${this.baseURI}/${parent_path}`
+            );
+            const contents = makeArray(body);
+
+            onCount(contents.length);
+
+            for (const item of contents) {
+                const contentWithPath: ContentItemWithPath = {
+                    ...item,
+                    full_path: parent_path
+                        ? `${parent_path}/${item.name}`
+                        : item.name
+                };
+                yield contentWithPath;
+            }
+        } catch (error) {
+            console.warn(
+                `Failed to get contents for path: ${parent_path}`,
+                error
+            );
+            onCount(0);
+        }
```

Comment:
```suggestion
    async *traverseChildren(
        parentPath = '',
        onCount: (total: number) => any = () => {}
    ): AsyncGenerator<ContentItemWithPath> {
            const { body } = await this.client.get<Content[]>(
                `${this.baseURI}/${parentPath}`
            );
            const contents = makeArray(body);

            onCount(contents.length);

            yield* content;
```
1. 每个 `content.path` 就是完整路径，不必画蛇添足
2. 错误保证抛出，让库调用方处理